application/lib/data_model.dart

```diff
@@ -309,18 +309,16 @@ class LandmarksPropertyValue extends PropertyValue {
   @override
   void registerChildren(Atom parent) {
     for (final Landmark landmark in value.where(Landmark.hasChild))
-      landmark.atom.registerFriend(parent);
+      landmark.atom!.registerFriend(parent);
   }
 
   @override
   void unregisterChildren(Atom parent) {
     for (final Landmark landmark in value.where(Landmark.hasChild))
-      landmark.atom.unregisterFriend(parent);
+      landmark.atom!.unregisterFriend(parent);
   }
-
   @override
-  Iterable<Atom> get children sync* { yield* value.where(Landmark.hasChild).map<Atom>((Landmark landmark) => landmark.atom); }
-
+  Iterable<Atom> get children sync* { yield* value.where(Landmark.hasChild).map<Atom?>((Landmark landmark) => landmark.atom) as Iterable<Atom>; }
```

Comment:
`as Iterable<Atom>` is terrifying. Can't we use `.map<Atom>` ?

Reply:
fixed